src/main/java/com/google/sps/servlets/PlaceGuideServlet.java

```diff
@@ -72,27 +74,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     String placeGuideQueryTypeString = request.getParameter(PLACE_GUIDE_QUERY_TYPE_PARAMETER);
     PlaceGuideQueryType placeGuideQueryType =
         PlaceGuideQueryType.valueOf(placeGuideQueryTypeString);
-    GeoPt northEastCorner = null;
-    GeoPt southWestCorner = null;
-    String creatorId = null;
-    if (placeGuideQueryType.requiresCoordinates()) {
-      String regionCornersString = request.getParameter(REGION_CORNERS_PARAMETER);
-      // On the client-side, the LatLngBound class's toUrlValue function will generate a string with
-      // the values being comma-separated. The string is parsed here.
-      String[] cornerCoordinates = regionCornersString.split(",");
-      southWestCorner =
-          new GeoPt(Float.parseFloat(cornerCoordinates[0]), Float.parseFloat(cornerCoordinates[1]));
-      northEastCorner =
-          new GeoPt(Float.parseFloat(cornerCoordinates[2]), Float.parseFloat(cornerCoordinates[3]));
-    }
-    if (placeGuideQueryType.requiresCoordinates()) {
-      creatorId = request.getParameter(CREATOR_ID_PARAMETER);
+    if (placeGuideQueryType == PlaceGuideQueryType.PLACE_GUIDE_WITH_ID) {
+      long placeGuideId = Long.parseLong(request.getParameter(PLACE_GUIDE_ID_PARAMETER));
+      PlaceGuide placeGuide = placeGuideRepository.getPlaceGuide(placeGuideId);
+      if (placeGuide == null) {
+        throw new IllegalArgumentException("There is no placeguide ith this id: " + placeGuideId);
+      }
+      String currentUserId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
+      PlaceGuideInfo placeGuideInfo = new PlaceGuideInfo(placeGuide, currentUserId);
+      response.setContentType("application/json;");
+      response.getWriter().println(convertToJsonUsingGson(placeGuideInfo));
```

Comment:
I would probably extract the code inside of the conditions to their own methods, to prevent this main `doGet` from growing too much.

Reply:
Done.